File: 프로그래머스/2/12951. JadenCase 문자열 만들기/JadenCase 문자열 만들기.java
import java.util.*;

class Solution {
    public String solution(String s) {
        StringBuilder out = new StringBuilder();
        boolean start = true;
        
        for(int i = 0; i < s.length(); i++) {
            char ch = s.charAt(i);
            
            if(ch == ' ') {
                start = true;
                out.append(ch);
            } else {
                if(start) {
                    out.append(Character.toUpperCase(ch));
                } else {
                    out.append(Character.toLowerCase(ch));
                }
                start = false;
            }
        }
        
        return out.toString();
    }
}

File: 프로그래머스/2/12951. JadenCase 문자열 만들기/JadenCase 문자열 만들기.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string s) {
    bool start = false;
    
    for(char& c : s) {
        if(c == ' ') {
            start = false;
            continue;
        }
        
        if(start) {
            c = tolower(c);
        } else {
            c = toupper(c);
            start = true;
        }
    }
    
    return s;
}